### PR TITLE
Fix/meta 3098 classification prop

### PR DESCRIPTION
**Pull request by BurakAltas**
## Change description

> User is informed that classification propagation is still being executed when there is a task in progress for the entity.

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review 

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java [hunk 1]
```diff
@@ -98,8 +98,30 @@ public List<AtlasTask> getPendingTasks() {
...
+        try {
+            AtlasGraphQuery query = graph.query()
+                    .has(Constants.TASK_TYPE_PROPERTY_KEY, Constants.TASK_TYPE_NAME)
+                    .has(Constants.TASK_STATUS, AtlasTask.Status.IN_PROGRESS)
+                    .orderBy(Constants.TASK_CREATED_TIME, AtlasGraphQuery.SortOrder.ASC);
+
+            Iterator<AtlasVertex> results = query.vertices().iterator();
+
+            while (results.hasNext()) {
+                AtlasVertex vertex = results.next();
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java:119]: add vertex != null  just to be safe

File: repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java [hunk 2]
```diff
@@ -98,8 +98,30 @@ public List<AtlasTask> getPendingTasks() {
...
         } catch (Exception exception) {
             LOG.error("Error fetching pending tasks!", exception);
-        } finally {
-            graph.commit();
+        }
+
+        return ret;
+    }
+
+    @GraphTransaction
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java:106]: I think no need to have @GraphTransaction here as this is the read operation
I am aware that this annotation is added at multiple other places where Read is performed, but I feel that we do not need to call GraphTransactionInterceptor, Let me know your thoughts

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationTask.java [hunk 3]
```diff
@@ -93,6 +92,8 @@ public AtlasTask.Status perform() throws AtlasBaseException {
         RequestContext.get().setUser(userName, null);
 
         try {
+            setStatus(IN_PROGRESS);
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationTask.java:95]: set task status to in progress in class AbstractTask itself
* This method does some extra work of removing task guid from entity vertex
* Adding to AbstractTask.run() will take effect for every run method which overrides this class e.g. MeaningsTask

**Review by nikhilbonte21 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationTask.java (hunk 3 above)
Comment (nikhilbonte21) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationTask.java:95]: Understood why it is called here, resolving this comment

Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/tasks/ClassificationTask.java:95]: addressed MeaningsTask

File: repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java (hunk 2 above)
Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java]: since we are not writing graphTransaction is redundant.

Comment (BurakAltas) [reply repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java]: addressed

**Review by nikhilbonte21 — APPROVED** [review]

**Pull request merged**